nodejs/test/layer.tap.js

```diff
@@ -34,11 +35,22 @@ tap.test('Layer tests', (t) => {
   })
 
   t.test('should wrap handler in transaction', async(t) => {
-    t.plan(2)
+    t.plan(4)
     helper.agent.on('transactionFinished', (tx) => {
       t.equal(tx.name, 'OtherTransaction/Function/testFn', 'transaction should be properly named')
     })
+
     const res = await handler({ key: 'value'}, { functionName: 'testFn'})
-    t.same(res, { statusCode: 200, body: 'response body' }, 'response should be correct')
+
+    t.same(res, { statusCode: 200, body: 'response body value' }, 'response should be correct')
+    // second run is to ensure that the promise is reused and no the whole process to require the handler
+
+    process.env = Object.assign(process.env, {
+      NEW_RELIC_LAMBDA_HANDLER: 'some_not/existing_path.handler'
+    })
```

Comment:
If this change to process.env has the intended effect, I'd expect the following test to fail.  Testing explicitly for that change with something like 
`t.same(process.env.NEW_RELIC_LAMBDA_HANDLER,
        'some_not/existing_path.handler', 'handler should have been changed')`
shows that the process.env.NEW_RELIC_LAMBDA_HANDLER isn't updated.